rpc/src/main.rs

```diff
@@ -1,33 +1,51 @@
-use std::io::Write;
+use std::io;
 
-use anyhow::Result;
-use clap::Parser;
-use cli::Commands;
-use rpc::{fetch_prover_input, FetchProverInputRequest};
+use alloy::{providers::RootProvider, rpc::types::eth::BlockId};
+use clap::{Parser, ValueHint};
+use tracing_subscriber::{prelude::*, EnvFilter};
+use url::Url;
 
-mod cli;
-mod init;
-mod rpc;
+#[derive(Parser)]
+pub enum Args {
+    /// Fetch and generate prover input from the RPC endpoint
+    Fetch {
+        /// The RPC URL
+        #[arg(short = 'u', long, value_hint = ValueHint::Url)]
+        rpc_url: Url,
+        /// The block number
+        #[arg(short, long)]
+        block_number: BlockId,
+        /// The checkpoint block number
+        // TODO(aatifsyed): impl Display for alloy::..::BlockId
```

Comment:
For this PR?

Reply:
WIBNI if BlockId: Display, but that's a change against alloy, which I'll open at some point